src/azure-cli-core/azure/cli/core/commands/validators.py

```diff
@@ -84,12 +85,26 @@ def get_default_location_from_resource_group(cmd, namespace):
 def validate_file_or_dict(string):
     import os
     string = os.path.expanduser(string)
-    if os.path.exists(string):
-        from azure.cli.core.util import get_file_json
-        return get_file_json(string)
-
-    from azure.cli.core.util import shell_safe_json_parse
-    return shell_safe_json_parse(string)
+    try:
+        if os.path.exists(string):
+            from azure.cli.core.util import get_file_json
+            # Failure point 1: 'string' is a valid file path, but the file contains invalid JSON string
+            # ex has no recommendation
+            return get_file_json(string)
+
+        from azure.cli.core.util import shell_safe_json_parse
+        # Failure point 2:
+        # - 'string' is a non-existing file path
+        # - 'string' is an invalid JSON string
+        # ex has recommendations for shell interpretation
+        return shell_safe_json_parse(string)
```

Comment:
> - 'string' is a non-existing file path

If the value entered by users ends with `.json` as the suffix, could we think that the users entered a non-existent JSON file, and then throw an exception to prompt that the JSON file does not exist?

Reply:
Maybe the user simply wants to specify `myfile.json` as a JSON string? We'd better not to assume `*.json` must be a JSON file.

Reply:
As discussed offline, if users want to pass `myfile.json` as a JSON string, they should specify `'"myfile.json"'`. A string ends with `*.json` most likely is a JSON file.

Reply:
Yes. I agree. When the user passes in the value ending with `.json`, most of cases are because the user wants to pass a JSON file, and a few cases are because of the user's wrong input or random testing.
Therefore, it may be more reasonable for us to focus on the major cases.